cmd/node-cache/main.go

```diff
@@ -176,28 +185,35 @@ func (c *cacheApp) parseAndValidateFlags() error {
 func (c *cacheApp) runChecks() {
 	for _, rule := range c.iptablesRules {
 		exists, err := c.iptables.EnsureRule(utiliptables.Prepend, rule.table, rule.chain, rule.args...)
-		if !exists {
-			if err != nil {
-				cache.teardownNetworking()
-				clog.Fatalf("Failed to add back non-existent rule %v", rule)
-			}
+		if exists {
+			// debug messages can be printed by including "debug" plugin in coreFile.
+			clog.Debugf("IP table rule (table %q, chain %q) already exists", rule.table, rule.chain)
+			continue
+		} else if err == nil {
 			clog.Infof("Added back nonexistent rule - %v", rule)
+			continue
 		}
-		if err != nil {
-			clog.Errorf("Failed to check rule %v - %s", rule, err)
+		// if we got here, either iptables check failed or adding rule back failed.
+		if isLockedErr(err) {
+			clog.Infof("Failed to check/add back rule %v, due to xtables lock in use, retrying in %v seconds", rule, c.params.interval)
+			setupErrCount.WithLabelValues("iptables_lock_error").Inc()
+		} else {
+			clog.Errorf("Failed to add back non-existent rule %v - %s", rule, err)
+			setupErrCount.WithLabelValues("iptables_err").Inc()
 		}
 	}
 
 	exists, err := c.netifHandle.EnsureDummyDevice(c.params.intfName)
 	if !exists {
 		if err != nil {
-			cache.teardownNetworking()
-			clog.Fatalf("Failed to add back non-existent interface %s", c.params.intfName)
+			clog.Errorf("Failed to add back non-existent interface %s: %s", c.params.intfName, err)
+			setupErrCount.WithLabelValues("intf_add_err").Inc()
```

Comment:
can we change the metric to interface_add_err?

Reply:
Fixed.